include/engine/data_watchdog.hpp

```diff
@@ -43,18 +43,24 @@ class DataWatchdogImpl<AlgorithmT, datafacade::ContiguousInternalMemoryDataFacad
             boost::interprocess::scoped_lock<mutex_type> current_region_lock(barrier.get_mutex());
 
             auto &shared_register = barrier.data();
-            auto region_id = shared_register.Find(dataset_name + "/data");
-            if (region_id == storage::SharedRegionRegister::INVALID_REGION_ID)
+            auto static_region_id = shared_register.Find(dataset_name + "/static");
```

Comment:
we could replace `auto` with `storage::SharedRegionRegister::RegionID` 🤷‍♀️ 
code style nitpick and up to you whether you want to pick it up or not 😂

Reply:
I usually prefer the short version, since it makes changes to the code easier. For example, when renaming the type `storage::SharedRegionRegister::RegionID` it should not break the code in too many places.

I found that in most cases it is good practice to write your code type-safe but also type-agnostic, meaning the type information is used by the compiler to find errors, but when reading the code you don't really have to care about the types that much. I think the short rule was: "`auto` whenever possible".

Reply:
yea. i can live with both. they have their advantages and disadvantages 👍

Reply:
@TheMarex sounds like Swift :-)